packages/resource-deployment/scripts/process-utilities.sh

```diff
@@ -12,20 +12,69 @@ function waitForProcesses() {
     for pid in "${!list}"; do
         echo "Waiting for process with pid $pid"
         wait $pid
-        echo "Process with pid $pid exited"
+        local pExitCode=$?
+        echo "Process with pid $pid exited with exit code $pExitCode"
+
+        if [[ $pExitCode != 0 ]]; then
+            echo "Process - $pid failed with exit code $pExitCode. Killing current proccess."
+            exit $pExitCode
+        fi
+
     done
 }
 
-function runInParallel() {
-    local processPaths=$1
+function runCommandsWithoutSecretsInParallel {
+    local commands=$1
     local -a parallelizableProcesses
 
-    list="$processPaths[@]"
-    for processPath in "${!list}"; do
-        . "${processPath}" &
-        echo "Created process with pid $! for process path - $processPath"
+    local list="$commands[@]"
+    for command in "${!list}"; do
+        eval "$command" &
+        echo "Created process with pid $! for command - $command"
         parallelizableProcesses+=("$!")
     done
 
     waitForProcesses parallelizableProcesses
+} 
+
+function sendSignalToProcessIfExists {
+    local currentPid=$1	
+    local signal=$2
+
+    if [[ -z $currentPid ]]; then	
+        return	
+    fi	
+
+    if kill -0 $currentPid > /dev/null 2>&1; then	
+        kill $signal $currentPid
+    fi
+}
+
+function killWithDecendentsIfProcessExists ()	
+{	
+    local currentPid=$1	
+
+    if [[ -z $currentPid ]]; then	
+        return	
+    fi	
+
+    if kill -0 $currentPid > /dev/null 2>&1; then	
+        echo "stopping process $currentPid"	
+        sendSignalToProcessIfExists $currentPid	-STOP
+       
+        killDescendentProcesses  $currentPid
+       
+        sendSignalToProcessIfExists $currentPid	-9
```

Comment:
seems tab used
add comment what is signal used

Reply:
not sure why it shows like this. but there is no tab. just a single space